geoutils/raster/raster.py

```diff
@@ -2464,15 +2480,29 @@ def icrop(
         if len(bbox) != 4:
             raise ValueError("bbox must be a list or tuple of four integers: (xmin, ymin, xmax, ymax).")
 
-        # Convert pixel coordinates to georeferenced coordinates using the transform
-        bottom_left = self.transform * (bbox[0], self.height - bbox[1])
-        top_right = self.transform * (bbox[2], self.height - bbox[3])
+        # Crop the raster data based on the given bounding box (bbox), limit the range to the borders of the raster
+        crop_img = self.data[..., bbox[0] : min(self.height, bbox[2]), bbox[1] : min(bbox[3], self.width)]
+
+        # Create a new affine transform that corresponds to the cropped area
```

Comment:
Took me a little while to understand why this transform looked like a translation!
When we crop within the extent, the new array size determines the new height/width, so it is only a translation for the affine transform!
In this case, let's use our tested `_translate` function to make it easier to read/debug for future us :wink::
https://github.com/GlacioHack/geoutils/blob/b4da49b8e5ff26f3120085c01cbaedbc1ee1b10b/geoutils/raster/geotransformations.py#L533